pkg/kubelet/cm/cpumanager/topology/topology_test.go

```diff
@@ -33,6 +33,41 @@ func Test_Discover(t *testing.T) {
 		want        *CPUTopology
 		wantErr     bool
 	}{
+		{
+			name: "EmptyUncoreCache",
+			machineInfo: cadvisorapi.MachineInfo{
+				NumCores:   8,
+				NumSockets: 1,
+				Topology: []cadvisorapi.Node{
+					{Id: 0,
+						Cores: []cadvisorapi.Core{
+							{SocketID: 0, Id: 0, Threads: []int{0, 4}, UncoreCaches: []cadvisorapi.Cache{}},
+							{SocketID: 0, Id: 1, Threads: []int{1, 5}, UncoreCaches: []cadvisorapi.Cache{}},
+							{SocketID: 0, Id: 2, Threads: []int{2, 6}, UncoreCaches: []cadvisorapi.Cache{}},
+							{SocketID: 0, Id: 3, Threads: []int{3, 7}, UncoreCaches: []cadvisorapi.Cache{}},
+						},
+					},
+				},
+			},
+			want: &CPUTopology{
+				NumCPUs:        8,
+				NumSockets:     1,
+				NumCores:       4,
+				NumNUMANodes:   1,
+				NumUncoreCache: 1,
+				CPUDetails: map[int]CPUInfo{
+					0: {CoreID: 0, SocketID: 0, NUMANodeID: 0, UncoreCacheID: 0},
+					1: {CoreID: 1, SocketID: 0, NUMANodeID: 0, UncoreCacheID: 0},
+					2: {CoreID: 2, SocketID: 0, NUMANodeID: 0, UncoreCacheID: 0},
+					3: {CoreID: 3, SocketID: 0, NUMANodeID: 0, UncoreCacheID: 0},
+					4: {CoreID: 0, SocketID: 0, NUMANodeID: 0, UncoreCacheID: 0},
+					5: {CoreID: 1, SocketID: 0, NUMANodeID: 0, UncoreCacheID: 0},
```

Comment:
we also need a test for `nil` uncore cache, when the `UncoreCaches` slice is `nil`